packages/amplify_native_legacy_wrapper/ios/Classes/AmplifyNativeLegacyWrapperPlugin.swift

```diff
@@ -0,0 +1,57 @@
+import Flutter
+import AmplifyPlugins
+import AWSPluginsCore
+import Amplify
+import UIKit
+
+public class AmplifyNativeLegacyWrapperPlugin: NSObject, FlutterPlugin, LegacyNativePlugin {
+    
+    private let authPlugin: AWSCognitoAuthPlugin = AWSCognitoAuthPlugin()
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let api = AmplifyNativeLegacyWrapperPlugin()
+        let binaryMessenger = registrar.messenger()
+        LegacyNativePluginSetup.setUp(binaryMessenger: binaryMessenger, api: api)
+    }
+    
+    func configure(config: String, completion: @escaping (Result<Void, Error>) -> Void) {
+        let decoder = JSONDecoder()
+        do {
+            let sourceData = config.data(using: .utf8)
+            let decodedObject = try decoder.decode(JSONValue.self, from: sourceData!)
+            let authConfig = decodedObject.value(at: "auth.plugins.awsCognitoAuthPlugin")
+            try authPlugin.configure(using: authConfig)
+            return completion(Result.success(()))
```

Comment:
Swift convention is to omit type when it can be inferred

```suggestion
            return completion(.success(()))
```